pybandits/strategy.py

```diff
@@ -167,7 +196,56 @@ def compare_best_actions(self, actions: Dict[ActionId, Beta]) -> float:
         return pvalue
 
 
-class CostControlBandit(Strategy):
+class CostControlStrategy(Strategy, ABC):
+    """
+    Cost Control (CC) strategy for multi-armed bandits.
+
+    Bandits are extended to include a control of the action cost. Each action is associated with a predefined "cost".
+    """
+
+    @classmethod
+    @validate_call
+    def _reduce(cls, p_of_action: Union[Probability, List[Probability]]):
```

Comment:
reduce is a very specific operation in computer science. 
I would create a reduce function (if doesn't already exist in Python) in utils 
`def reduce(acc: list, op: callable)` and I would call this method something like mean_of, which calls the actual reduce with either `p_of_action` or `[p_of_action]` and as operation a function that compute the mean

Reply:
I took your solution partially. I made it like this:
`    def _average(cls, p_of_action: Union[Probability, List[Probability]]):
        return np.mean(p_of_action)`
I did not see the point of defining a general reduce function at the moment. And of course, there is the functools' reduce function.